### PR TITLE
[protobuf] fix pc library reference

### LINKED ISSUE #38757: [protobuf] pc files reference wrong library on windows
`Libs: "-L${libdir}" -lprotobuf`

However on windows protobuf has a lib prefix. So the correct way to reference is `-llibprotobuf`

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix https://github.com/microsoft/vcpkg/issues/38757

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~~SHA512s are updated for each updated download.~~
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/protobuf/portfile.cmake [hunk 1]
```diff
@@ -121,12 +121,28 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
...
+    set(debug_file "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/${package}.pc")
+    set(release_file "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/${package}.pc")
+
+    if(EXISTS "${release_file}")
+        vcpkg_replace_string(${release_file} "-l${package}" "-llib${package}")
     endif()
+
+    if(EXISTS "${debug_file}")
+        vcpkg_replace_string(${debug_file} "-l${package}" "-llib${package}d")
+    endif()
```
Comment (dg0yt) [inline ports/protobuf/portfile.cmake:135]: AFAICT `lib` only needs to be prepended for `VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW`.

Comment (dg0yt) [reply ports/protobuf/portfile.cmake:135]: And only for `protobuf*`, not for extra link libs if they exist.

File: ports/protobuf/portfile.cmake [hunk 2]
```diff
@@ -121,13 +121,26 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
...
+
+    if(EXISTS "${debug_file}")
+        vcpkg_replace_string(${debug_file} "-l${package}" "-llib${package}d")
     endif()
-endforeach()
+endfunction()
+
+set(packages protobuf protobuf-lite)
+if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
```
Comment (dg0yt) [inline ports/protobuf/portfile.cmake:139]: This now no longer appends the `d` suffix for most platforms. 😠

**Review by Cheney-W — DISMISSED** [review]

**Comment by Neumann-A** [comment]
@Cheney-W: Why do you approve this with https://github.com/microsoft/vcpkg/pull/38822#discussion_r1609704437 being unfixed?

**Comment by Cheney-W** [comment]
Sorry, I didn't refresh the page when I added the approval, so I didn't see dg0yt's comment at the time. I saw it only after I approved.

**Review by Cheney-W — APPROVED** [review]

**Comment by BillyONeal** [comment]
@Cheney-W Thanks for the review
@dg0yt Thanks for the review and spotting a bug
@FrankXie05 Thanks for the fix!

**Pull request merged**